ecosystem/python/sdk/aptos_sdk/bcs.py

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This is a simple BCS serializer and deserializer. Learn more at https://github.com/diem/bcs
+"""
+
+from __future__ import annotations
+
+import io
+import typing
+import unittest
+
+MAX_U8 = 2**8 - 1
+MAX_U16 = 2**16 - 1
+MAX_U32 = 2**32 - 1
+MAX_U64 = 2**64 - 1
+MAX_U128 = 2**128 - 1
+
+
+class Deserializer:
+    _input: io.BytesIO
+    _length: int
+
+    def __init__(self, data: bytes):
+        self._length = len(data)
+        self._input = io.BytesIO(data)
+
+    def remaining(self) -> int:
+        return self._length - self._input.tell()
+
+    def bool(self) -> bool:
+        value = int.from_bytes(self._read(1), byteorder="little", signed=False)
+        if value == 0:
+            return False
+        elif value == 1:
+            return True
+        else:
+            raise Exception("Unexpected boolean value: ", value)
+
+    def bytes(self) -> bytes:
+        return self._read(self.uleb128())
+
+    def fixed_bytes(self, length: int) -> bytes:
+        return self._read(length)
+
+    def map(
+        self,
+        key_decoder: typing.Callable[[Deserializer], typing.Any],
+        value_decoder: typing.Callable[[Deserializer], typing.Any],
+    ) -> Dict[typing.Any, typing.Any]:
+        length = self.uleb128()
+        values = {}
+        while len(values) < length:
+            key = key_decoder(self)
+            value = value_decoder(self)
+            values[key] = value
+        return values
```

Comment:
See my comment on Serializer's map. We need to consider ordering?

Reply:
for deserialization, we could (and maybe should) check